pkg/providers/instance/instance.go

```diff
@@ -700,6 +702,38 @@ func (p *DefaultProvider) handleResponseErrors(ctx context.Context, instanceType
 
 		return fmt.Errorf("unable to allocate resources in the selected zone (%s). (will try a different zone to fulfill your request)", zone)
 	}
+	// TODO - should we be handling this error? Block specific VM Size in all zones?
```

Comment:
I wasn't sure if this is an error we want to be reacting to here. I added it just in case we want to keep it, docs I based this on: https://learn.microsoft.com/en-us/troubleshoot/azure/cloud-services/extended/availability/allocation-failure 
however there is also AKS specific doc which talks about this error only in context of dedicated hosts, which AFAIK we don't support here atm? https://learn.microsoft.com/en-us/troubleshoot/azure/azure-kubernetes/create-upgrade-delete/error-code-zonalallocationfailed-allocationfailed

Reply:
I see no harm keeping it in. We can also check whether this has been ever observed.